tests/ui/use/use-path-segment-kw.rs

```diff
@@ -0,0 +1,177 @@
+//@ edition: 2021
+
+macro_rules! macro_dollar_crate {
+    ($m: ident) => {
+        use $crate; //~ ERROR `$crate` may not be imported
+        pub use $crate as _dollar_crate; // Good
+
+        use ::$crate; //~ ERROR `$crate` in paths can only be used in start position
+        use ::$crate as _dollar_crate2; //~ ERROR `$crate` in paths can only be used in start position
+        use ::{$crate}; //~ ERROR `$crate` in paths can only be used in start position
+        use ::{$crate as _nested_dollar_crate2}; //~ ERROR `$crate` in paths can only be used in start position
+
+        use $m::$crate; //~ ERROR `$crate` in paths can only be used in start position
+        use $m::$crate as _dollar_crate3; //~ ERROR `$crate` in paths can only be used in start position
+        use $m::{$crate}; //~ ERROR `$crate` in paths can only be used in start position
+        use $m::{$crate as _nested_dollar_crate3}; //~ ERROR `$crate` in paths can only be used in start position
+
+        use crate::$crate; //~ ERROR `$crate` in paths can only be used in start position
+        use crate::$crate as _dollar_crate4; //~ ERROR `$crate` in paths can only be used in start position
+        use crate::{$crate}; //~ ERROR `$crate` in paths can only be used in start position
+        use crate::{$crate as _nested_dollar_crate4}; //~ ERROR `$crate` in paths can only be used in start position
+
+        use super::$crate; //~ ERROR `$crate` in paths can only be used in start position
+        use super::$crate as _dollar_crate5; //~ ERROR `$crate` in paths can only be used in start position
+        use super::{$crate}; //~ ERROR `$crate` in paths can only be used in start position
+        use super::{$crate as _nested_dollar_crate5}; //~ ERROR `$crate` in paths can only be used in start position
+
+        use self::$crate; //~ ERROR `$crate` in paths can only be used in start position
```

Comment:
It makes sense to add the opposite order combinations as well (`$crate::self`, `$crate::super`, etc) because they are not tested below.